data_structures/linked_list/middle_element_of_linked_list.py

```diff
@@ -0,0 +1,60 @@
+class Node:
+    def __init__(self, data:int) -> int:  
+        self.data = data  
+        self.next = None
+
+class LinkedList:
+    def __init__(self): 
+        self.head = None
+
+    def push(self, new_data:int) -> int:
+        new_node = Node(new_data) 
+        new_node.next = self.head 
+        self.head = new_node
+        return self.head.data
+
+    def middle_element(self) -> int:
+        '''
+        >>> link = LinkedList()
+        >>> link.push(5)
```

Comment:
One more situation that we should include in our tests...  What happens when we call middle_element() on an __empty__ LinkedList?

```suggestion
        >>> link = LinkedList()
        >>> link.middle_element()
        Traceback (most recent call last):
        ...
        AttributeError: 'NoneType' object has no attribute 'data'
        >>> link.push(5)
        >>> link.push(5)
```